fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckPreferencesMiddleware.kt

```diff
@@ -103,6 +126,53 @@ class ReviewQualityCheckPreferencesMiddleware(
                     }
                 }
             }
+
+            ReviewQualityCheckAction.ExpandCollapseHighlights -> {
+                appStore.dispatchShoppingAction(
+                    reviewQualityCheckState = store.state,
+                    action = { productPageUrl, optedIn ->
+                        HighlightsCardExpanded(
+                            productPageUrl = productPageUrl,
+                            expanded = optedIn.isHighlightsExpanded,
+                        )
+                    },
+                )
+            }
+
+            ReviewQualityCheckAction.ExpandCollapseInfo -> {
+                appStore.dispatchShoppingAction(
+                    reviewQualityCheckState = store.state,
+                    action = { productPageUrl, optedIn ->
+                        InfoCardExpanded(
+                            productPageUrl = productPageUrl,
+                            expanded = optedIn.isInfoExpanded,
+                        )
+                    },
+                )
+            }
+
+            ReviewQualityCheckAction.ExpandCollapseSettings -> {
+                appStore.dispatchShoppingAction(
+                    reviewQualityCheckState = store.state,
+                    action = { productPageUrl, optedIn ->
+                        SettingsCardExpanded(
+                            productPageUrl = productPageUrl,
+                            expanded = optedIn.isSettingsExpanded,
+                        )
+                    },
+                )
+            }
+        }
+    }
+
+    private fun Store<AppState, AppAction>.dispatchShoppingAction(
+        reviewQualityCheckState: ReviewQualityCheckState,
+        action: (productPageUrl: String, optedIn: OptedIn) -> ShoppingAction,
```

Comment:
why the lambda instead of just passing a `ShoppingAction` as the parameter?

Reply:
The `ShoppingAction` needs `productPageUrl` and the right expanded boolean to be constructed. For instance,